ql/src/test/org/apache/hadoop/hive/ql/io/orc/TestInputOutputFormat.java

```diff
@@ -2875,10 +2875,8 @@ public void testSplitGenReadOpsLocalCache() throws Exception {
         readOpsDelta = statistics.getReadOps() - readOpsBefore;
       }
     }
-    // call-1: listLocatedStatus - mock:/mocktbl
-    // call-2: check existence of side file for mock:/mocktbl/0_0
-    // call-3: check existence of side file for  mock:/mocktbl/0_1
-    assertEquals(3, readOpsDelta);
+    // call-1: getAcidState - mock:/mocktbl
```

Comment:
Why is this change? It seems, test is changed to make it work. Any reason?

Reply:
This PR targets to skip flush length check for external tables. The reason that some of the queries on external tables took more time was because Hive used to check for flush length files which are only present for Acid tables. In the first call when it concludes that it is an external table, it skips checking for side files hence the total number of calls reduces from 3 to 1.